utils/src/merkle_tree/optimal_merkle_tree.rs

```diff
@@ -316,6 +322,66 @@ where
         }
         Ok(())
     }
+
+    /// Update hashes after some leaves have been set or updated
+    /// index - first leaf index (which has been set or updated)
+    /// length - number of elements set or updated
+    fn update_hashes(&mut self, index: usize, length: usize) -> Result<()> {
+        // parent depth & index (used to store in the tree)
+        let mut parent_depth = self.depth - 1; // tree depth (or leaves depth) - 1
+        let mut parent_index = index >> 1;
+        let mut parent_index_bak = parent_index;
```

Comment:
what does `_bak` mean?

Reply:
_bak is a shortcut for backup because here we "backup" (or save) the parent index value. Should we rename it if it is not clear enough?

Reply:
got it, no need to change